datafusion/core/src/execution/context/mod.rs

```diff
@@ -114,17 +114,20 @@ impl DataFilePaths for &String {
     }
 }
 
-impl<P> DataFilePaths for Vec<P>
-where
-    P: AsRef<str>,
-{
+impl DataFilePaths for Vec<&str> {
```

Comment:
I think the previous implementation would have allowed `Vec<String>` (and anything else) that implemented `AsRef<str>` as well and after this change `Vec<String>` 

I wonder if we could change the signature to something like

```rust
impl<P> DataFilePaths for Vec<P>
where
    P: Into<ListingTableUrl>,
```

Which should still work for anything that allows `AsRef<str>` but also allow `Vec<ListingTableUrl>` 🤔

Reply:
Hmm I think this may fail since `AsRef<str>` doesn't imply`Into<ListingTableUrl>`. It has to go through a `ListingTableUrl::parse`, which is fallible. 

I'll give it some more thought and play with the types. Perhaps I'll try to change how `SessionContext::read_parquet` handles `AsRef<str>` and `ListingTableUrl`.

Reply:
Thanks @OussamaSaoudi

Reply:
I'm not sure there's a clean way to implement this change without a breaking existing code.  Here are the two things I tried: 
- I briefly considered  changing the `as_str` impl of `ListingTableChanges` to make it succeed with a `parse`, but this is both a breaking change and pretty silly.
- I tried using a blanket implement using TryInto, and use `ListingTableUrl::parse` as a `TryInto<ListingTableUrl>` impl. However I hit some issues:
```rust
impl<P> DataFilePaths for Vec<P>
where P: TryInto<ListingTableUrl, Error=DataFusionError>
```
The problem is the associated type for `TryInto`. A single type (ex`ListingTableUrl`) can implement `TryInto<ListingTableUrl, Error=Infallible>` , or `TryInto<ListingTableUrl, Error=DataFusionError>` (ex: through `parse`). This of course leads to conflicting types for my blanket impl approach.
- I looked through `SessionContext::read_parquet`. We can't change the use of `DataFilePaths::to_urls()` since that's baked into the API:
```rust
    pub async fn read_parquet<P: DataFilePaths>(
        &self,
        table_paths: P,
        options: ParquetReadOptions<'_>,
```
I would've preferred to separate the concerns of parsing and fetching parquet. An approach that parses strings into `ListingTableUrl` before calling `read_parquet` would be my preferred solution.

If DataFusion is not looking to change these APIs (understandably), I can go ahead and close this PR :)